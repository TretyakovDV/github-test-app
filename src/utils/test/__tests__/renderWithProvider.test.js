import React from 'react';
import { inject, observer } from 'mobx-react';
import renderWithProvider from '../renderWithProvider';

describe('renderWithProvider', () => {
  it('matches snapshot', () => {
    const Component = inject('test')(observer(({ test }) => <div data-testid="test">{test.label}</div>));

    const store = {
      test: {
        label: 'label',
      },
    };

    const { asFragment } = renderWithProvider(<Component />, store);
    expect(asFragment()).toMatchSnapshot();
  });

  it('passed store value', () => {
    const Component = inject('test')(observer(({ test }) => <div data-testid="test">{test.label}</div>));

    const store = {
      test: {
        label: 'label',
      },
    };

    const { getByTestId } = renderWithProvider(<Component />, store);
    expect(getByTestId('test')).toHaveTextContent('label');
  });
});
