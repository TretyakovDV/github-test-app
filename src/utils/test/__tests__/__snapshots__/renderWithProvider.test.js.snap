// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderWithProvider matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="test"
  >
    label
  </div>
</DocumentFragment>
`;
